botocore/vendored/requests/packages/urllib3/__init__.py

```diff
@@ -7,63 +7,4 @@
 __version__ = '1.10.4'
 
 
-from .connectionpool import (
-    HTTPConnectionPool,
-    HTTPSConnectionPool,
-    connection_from_url
-)
-
 from . import exceptions
```

Comment:
This seems odd:
* vendored `requests` are removed
* vendored `urllib3` is removed
* but vendored `urllib3.exceptions` remains?

Reply:
re: "for those who used these features"
I think perhaps this can be worked around by e.g.:
```
# vendored/__init__.py
import requests
```
which would ensure that user can still `from botocore.vendored.requests.urllib3.exceptions import X`
In any case, if someone were to catch e.g. `HTTPError`, it needs to be same `HTTPError` is thrown in the dependencies, right?

Reply:
It is a little strange, but let me provide a little more context on this. In earlier versions of `botocore` we leaked exceptions from `requests` which meant that in some cases users may have needed to use an import an exception class similar to what you show:
```
from botocore.vendored.requests.urllib3.exceptions import X
```
In the migration to `urllib3` we generally followed the exception structure in `requests`/`urllib3` and mapped them to our [own exception classes](https://github.com/boto/botocore/blob/48de08925d596172211dc5ae5642ef525845946a/botocore/exceptions.py#L79-L119). Leaving the exception classes is simply to ensure users using a line similar to the above will continue to work (as closely as possible) and avoid a hard import error.

We have considered the latter suggestion but as we no longer depend on `requests` it's a little more tricky than just doing the import as `requests` may or may not be present. We also have no control over the version of `requests` which leaves the possibility for something to change in `requests` breaking the SDK (though this is probably unlikely). I know that `requests` did a similar change when they removed their vendored `urllib3` but things are pretty different here since we no longer depend on `requests`.